akka-remote/src/main/scala/akka/remote/artery/ImmutableLongMap.scala

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import scala.annotation.tailrec
+import akka.util.OptionVal
+import scala.reflect.ClassTag
+import java.util.Arrays
+
+/**
+ * INTERNAL API
+ */
+private[akka] object ImmutableLongMap {
+  def empty[A >: Null](implicit t: ClassTag[A]): ImmutableLongMap[A] =
+    new ImmutableLongMap(Array.emptyLongArray, Array.empty)
```

Comment:
yeah, I was a bit uncertain about if that works for the class tag and the value array.
why is that not done in scala `Array.empty`?